soccer/gameplay/plays/testing/four_courners_pass.py

```diff
@@ -0,0 +1,201 @@
+import robocup
+import play
+import behavior
+import skills.move
+import skills.capture
+import tactics.coordinated_pass
+import constants
+import main
+import enum
+
+
+## A testing play to demonstrate our ability to pass and recieve balls
+# One robot will pursue the ball while three other robots will pass the ball amongst themselves
+# they can only pass the ball at the corners and recieve at the corners and they cannot move
+# through the center of the square
+class FourCornerPass(play.Play):
+	class State(enum.Enum):
+		## One robot goes captures the ball while the other two gets on corners
+		# pursuing robot does not move on the first instance
+		setup = 1
+
+		## The robots pass to each other and begin setting up the next pass
+		# while the chasing robot goes for the ball.
+		passing = 2
+
+	def __init__(self):
+		super().__init__(continuous=True)
+
+		# register states - they're both substates of "running"
+		self.add_state(FourCornerPass.State.setup,
+					   behavior.Behavior.State.running)
+		self.add_state(FourCornerPass.State.passing,
+					   behavior.Behavior.State.running)
+
+		self.add_transition(behavior.Behavior.State.start,
+							FourCornerPass.State.setup, lambda: True,
+							'immediately')
+		#transition when ball is captured
+		self.add_transition(
+			FourCornerPass.State.setup, FourCornerPass.State.passing, 
+			lambda: self.subbehavior_with_name('capture').state == behavior.Behavior.
+			State.completed , 'all subbehaviors completed')
+
+		#transition when ball is passed or failed to pass
+		self.add_transition(
+			FourCornerPass.State.passing, FourCornerPass.State.setup, 
+			lambda: self.subbehavior_with_name('passer').state == behavior.Behavior.State.completed
+			or self.subbehavior_with_name('passer').state == behavior.Behavior.
+			State.failed, 'all subbehaviors completed')
+
+		#change this to adjust the square size
+		variable_square = 5.5
+
+		#NEVER CHANGE THIS
+		length = constants.Field.Length
+		width = constants.Field.Width
+
+		#if the square size is smaller than the width shrink it down otherwise
+		#max value is the width of the field.
+		variable_square = min(variable_square, width, length)
+
+		#radius of robot
+		self.bot = constants.Robot.Radius
+		# the largest and smallest x and y postion possible for the square
+		self.max_x = (variable_square / 2 - self.bot * 3)
+		self.min_x = -self.max_x
+		self.max_y = length/2 + self.max_x
+		self.min_y = length/2 - self.max_x
+
+		# the four courners
+		self.square_points = [
+			robocup.Point(self.max_x, self.max_y),
+			robocup.Point(self.min_x, self.max_y),
+			robocup.Point(self.min_x, self.min_y),
+			robocup.Point(self.max_x, self.min_y)]
+
+		# speed of the hunting robots
+		self.normal_speed = 2.0
+		# speed of the defending robots can decrease value to make it easier for offense
+		self.defense_speed = variable_square/3.0
+
+		# picks the direction to pass to. TODO make actual smart pass selection
```

Comment:
I think you can remove this todo and either place it in the correct function or just leave it off

Reply:
Do TODO's come up in our code base?

Reply:
They stay and if you search for them, you can find them all, but there's no build stuff that warns or anything